codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/transformers/EventStreamNormalizer.kt

```diff
@@ -37,3 +80,40 @@ object EventStreamNormalizer {
             .build()
     }
 }
+
+fun OperationShape.operationErrors(model: Model): List<Shape> {
+    val operationIndex = OperationIndex.of(model)
+    return operationIndex.getErrors(this)
+        .filter { it.hasTrait<SyntheticOperationErrorTrait>() }
+}
+
+fun eventStreamErrors(model: Model, shape: Shape): Map<UnionShape, List<StructureShape>> {
+    return Walker(model).walkShapes(shape)
+        .filter { it is UnionShape && it.isEventStream() }
+        .map { it.asUnionShape().get() }
+        .associateWith { unionShape ->
+            unionShape.expectTrait<SyntheticEventStreamUnionTrait>().errorMembers
+                .map { model.expectShape(it.target).asStructureShape().get() }
+        }
+}
+
+fun UnionShape.inputEventStreamErrors(): List<Shape> {
```

Comment:
Should this be renamed to `eventStreamErrors` since it's the same logic for both inputs and outputs?